org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/EclipseProjectImporter.java

```diff
@@ -32,17 +35,32 @@
 import org.eclipse.jdt.core.IJavaProject;
 import org.eclipse.jdt.core.JavaCore;
 import org.eclipse.jdt.ls.core.internal.AbstractProjectImporter;
+import org.eclipse.jdt.ls.core.internal.IProjectImporter;
 import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
+import org.eclipse.jdt.ls.core.internal.ProjectUtils;
 
 public class EclipseProjectImporter extends AbstractProjectImporter {
 
-	private Collection<java.nio.file.Path> directories;
-
 	@Override
 	public boolean applies(IProgressMonitor monitor) throws CoreException {
 		if (directories == null) {
 			BasicFileDetector eclipseDetector = new BasicFileDetector(rootFolder.toPath(), DESCRIPTION_FILE_NAME)
 					.addExclusions("**/bin");//default Eclipse build dir
+			List<IProjectImporter> importers = new ArrayList<>(ProjectsManager.importers());
+			for (IProject project : ProjectUtils.getAllProjects(false)) {
```

Comment:
I'm wondering whether we can exclude these projects directly. These projects are probably from the importer with higher priority, if so, then importer.applies must return true. Or come from the projects cached from the last time the workspace was opened, if the project is already on `open` state, looks not necessary to scan it and reopen it via EclipseProjectImporter. Is there any missing corner case?

Reply:
You are right. Fixed.